generate/generate.go

```diff
@@ -135,150 +163,300 @@ func Load(cfg *config.Tree) ([]LoadResult, error) {
-	report := forEachStack(cfg, workingDir, func(
-		projmeta project.Metadata,
-		stack *stack.S,
-		globals *eval.Object,
-	) dirReport {
-		stackpath := stack.HostPath()
-		logger := log.With().
-			Str("action", "generate.Do()").
-			Stringer("stack", stack.Path()).
-			Logger()
+	stackReport := forEachStack(cfg, workingDir, doStackGeneration)
+	dirReport := forEachDir(cfg, doDirGeneration)
+	report := mergeReports(stackReport, dirReport)
+	return cleanupOrphaned(cfg, report)
+}
 
-		report := dirReport{}
+func doStackGeneration(
+	cfg *config.Tree,
+	projmeta project.Metadata,
+	stack *stack.S,
+	globals *eval.Object,
+) dirReport {
+	stackpath := stack.HostPath()
+	logger := log.With().
+		Str("action", "generate.doStackGeneration()").
+		Stringer("stack", stack.Path()).
+		Logger()
 
-		logger.Debug().Msg("generating files")
+	report := dirReport{}
 
-		asserts, err := loadAsserts(cfg, projmeta, stack, globals)
-		if err != nil {
-			report.err = err
-			return report
-		}
+	logger.Debug().Msg("generating files")
 
-		generated, err := loadStackCodeCfgs(cfg, projmeta, stack, globals)
-		if err != nil {
-			report.err = err
-			return report
-		}
+	asserts, err := loadAsserts(cfg, projmeta, stack, globals)
+	if err != nil {
+		report.err = err
+		return report
+	}
 
-		for _, gen := range generated {
-			asserts = append(asserts, gen.Asserts()...)
-		}
+	generated, err := loadStackCodeCfgs(cfg, projmeta, stack, globals)
+	if err != nil {
+		report.err = err
+		return report
+	}
 
-		errs := errors.L()
-		for _, assert := range asserts {
-			if !assert.Assertion {
-				assertRange := assert.Range
-				assertRange.Filename = project.PrjAbsPath(cfg.RootDir(), assert.Range.Filename).String()
-				if assert.Warning {
-					log.Warn().
-						Stringer("origin", assertRange).
-						Str("msg", assert.Message).
-						Stringer("stack", stack.Path()).
-						Msg("assertion failed")
-				} else {
-					msg := fmt.Sprintf("%s: %s", assertRange, assert.Message)
+	for _, gen := range generated {
+		asserts = append(asserts, gen.Asserts()...)
+	}
 
-					logger.Debug().Msgf("assertion failure detected: %s", msg)
+	err = handleAsserts(cfg.RootDir(), stack.HostPath(), asserts)
+	if err != nil {
+		report.err = err
+		return report
+	}
 
-					err := errors.E(ErrAssertion, msg)
-					errs.Append(err)
-				}
-			}
+	errsmap := checkFileConflict(generated)
+	if len(errsmap) > 0 {
+		errs := errors.L()
+		for _, err := range errsmap {
+			errs.Append(err)
 		}
+		report.err = errs.AsError()
+		return report
+	}
 
-		if err := errs.AsError(); err != nil {
-			report.err = err
-			return report
-		}
+	err = validateStackGeneratedFiles(cfg, stackpath, generated)
+	if err != nil {
+		report.err = err
+		return report
+	}
 
-		err = validateGeneratedFiles(cfg, stackpath, generated)
-		if err != nil {
-			report.err = err
-			return report
+	var removedFiles map[string]string
+
+	failureReport := func(r dirReport, err error) dirReport {
```

Comment:
this failureReport thing would not be necessary without the remove all files ugly hack, sorry for this x_x